src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -1190,4 +1197,34 @@ public InputStream getFileContent(BitbucketSCMFile file) throws IOException, Int
         }
         return content;
     }
+
+    @Restricted(ProtectedExternally.class)
+    protected CloseableHttpResponse executeMethod(CloseableHttpClient client, HttpRequestBase httpMethod) throws IOException, InterruptedException {
+        CloseableHttpResponse response = client.execute(httpMethod, context);
+        Instant start = Instant.now();
+        Instant forcedEnd = start.plus(API_RATE_LIMIT_MAX_SLEEP);
+        Duration sleepDuration = API_RATE_LIMIT_INITIAL_SLEEP;
+        while (response.getStatusLine().getStatusCode() == API_RATE_LIMIT_STATUS_CODE
+                && Instant.now().plus(sleepDuration).isBefore(forcedEnd)) {
+            try {
+                response.close();
+            } finally {
+                httpMethod.releaseConnection();
```

Comment:
To be honest, I do not know if this is needed, but it was in #483. I will investigate the relevant API in more detail tomorrow.

Reply:
So we do need to call `releaseConnection` to be able to reuse the request, but there is no need for a `try`/`finally` block since nothing is actually being cleaned up (`releaseConnection` just delegates to `reset`). Cleaned up in https://github.com/jenkinsci/bitbucket-branch-source-plugin/pull/581/commits/9d570222769daa61bee4262d9c5819cc4d9e8088.